needle-zendesk-connector/src/server/backend/connectors-backend.ts

```diff
@@ -0,0 +1,257 @@
+  if (!connectorDetails) {
+    throw new Error(`No Zendesk connector found for ID: ${connectorId}`);
+  }
+
+  const currentFiles = await getCurrentFiles(connectorId);
+  const currentTickets = currentFiles.filter(
+    (f) => f.type === "ticket" || f.type === "comments",
+  ) as DbFile[];
+  const currentArticles = currentFiles.filter(
+    (f) => f.type === "article",
+  ) as DbFile[];
+
+  const zendeskService = createZendeskService(
+    accessToken,
+    connectorDetails.subdomain,
+  );
+  const { tickets: liveTickets, articles: liveArticles } =
+    await zendeskService.searchByOrganization(
+      connectorDetails.orgId.toString(),
+      {
+        fetchTickets: connectorDetails.includeTickets,
+        fetchArticles: connectorDetails.includeArticles,
+      },
+    );
+
+  const ticketsDiff = liveTickets
+    ? computeDiff(currentTickets, liveTickets.items)
+    : { create: [], update: [], delete: currentTickets };
+  const articlesDiff = liveArticles
+    ? computeDiff(currentArticles, liveArticles.items)
+    : { create: [], update: [], delete: currentArticles };
+
+  const descriptor: ConnectorRunDescriptor = {
+    create: [],
+    update: [],
+    delete: [],
+  };
+  const fileIdMap = new Map<string, string>();
+
+  // Process creates
+  for (const ticket of ticketsDiff.create) {
+    if ("subject" in ticket) {
+      const ticketFileId = createNeedleFileId();
+      const commentsFileId = createNeedleFileId();
+      fileIdMap.set(`ticket-${ticket.id}`, ticketFileId);
+      fileIdMap.set(`comments-${ticket.id}`, commentsFileId);
+
+      descriptor.create.push(
+        {
+          id: ticketFileId,
+          url: ticket.url,
+          type: "text/plain",
+        },
+        {
+          id: commentsFileId,
+          url: ticket.url.replace(".json", "/comments.json"),
+          type: "text/plain",
+        },
+      );
+    }
+  }
+
+  for (const article of articlesDiff.create) {
+    if ("title" in article) {
+      const articleFileId = createNeedleFileId();
+      fileIdMap.set(`article-${article.id}`, articleFileId);
+
+      descriptor.create.push({
+        id: articleFileId,
+        url: article.html_url,
+        type: "text/html",
+      });
+    }
+  }
+
+  // Process updates
+  for (const item of [...ticketsDiff.update, ...articlesDiff.update]) {
+    const dbFiles = currentFiles.filter((f) => f.originId === item.id);
+    for (const dbFile of dbFiles) {
+      if (dbFile.ndlFileId) {
+        descriptor.update.push({ id: dbFile.ndlFileId });
+      }
+    }
+  }
+
+  // Process deletes
+  const filesToDelete = currentFiles.filter((file) =>
+    file.type === "ticket" || file.type === "comments"
+      ? ticketsDiff.delete.some((t) => t.originId === file.originId)
+      : articlesDiff.delete.some((a) => a.originId === file.originId),
+  );
+
+  for (const file of filesToDelete) {
+    if (file.ndlFileId) {
+      descriptor.delete.push({ id: file.ndlFileId });
+    }
+  }
+
+  try {
```

Comment:
I think we can just omit the try catch. In the current usage only benefit is the extra line of log. I'm ok seeing just the exception.

Reply:
Sure, done!